src/basic/random-util.c

```diff
@@ -30,183 +30,140 @@
 #include "missing_syscall.h"
 #include "parse-util.h"
 #include "random-util.h"
-#include "siphash24.h"
+#include "sha256.h"
 #include "time-util.h"
 
-static bool srand_called = false;
+static void fallback_random_bytes(void *p, size_t n) {
+        static thread_local uint64_t fallback_counter = 0;
+        struct {
+                char label[32];
+                uint64_t call_id, block_id;
+                usec_t stamp_mono, stamp_real;
+                pid_t pid, tid;
+                uint8_t auxval[16];
+        } state = {
+                .label = "systemd fallback random bytes v1",
```

Comment:
What is this for?

Reply:
please add a comment, as it's not immediately obvious

Reply:
Done.

Reply:
Yes. The idea is to separate the domain definitively so, rather than probably so. For this reason, most crypto protocols (TLS, Signal, WireGuard, etc) use labels at the beginning of hash chains. I mentioned "hygiene" in my last message, because this is something I do out of habit. We could _probably_ reason that it's not necessary and that this is overly paranoid. But it's cheap enough to do, for code that will never be used anyway, and it's one less thing to think about.